lntest/itest/lnd_test.go

```diff
@@ -13966,19 +13966,26 @@ func sendAndAssertSuccess(t *harnessTest, node *lntest.HarnessNode,
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
 	defer cancel()
 
-	stream, err := node.RouterClient.SendPaymentV2(ctx, req)
-	if err != nil {
-		t.Fatalf("unable to send payment: %v", err)
-	}
+	var result *lnrpc.Payment
+	err := wait.NoError(func() error {
+		stream, err := node.RouterClient.SendPaymentV2(ctx, req)
+		if err != nil {
+			return fmt.Errorf("unable to send payment: %v", err)
+		}
 
-	result, err := getPaymentResult(stream)
-	if err != nil {
-		t.Fatalf("unable to get payment result: %v", err)
-	}
+		result, err = getPaymentResult(stream)
```

Comment:
Shouldn't this part be in a separate `wait.NoError`?

Reply:
Not in this specific case. The flake wasn't that the result wasn't returned in time but that the payment couldn't be made because the router was still thinking it didn't have any channels/balance to work with. So we wait until the router doesn't return "not enough local balance" anymore.